### ISSUE TITLE
No battery values

### ISSUE THREAD

**Issue by bartpaeleman** [muppet3000/homeassistant-grott#10]
Got my GROTT docker container set up, and it is talking with my MQTT docker, sending values into my HA container through the Grott integration from Chris. I still have the old integration active too so I can update over all of my automations but I noticed that there are several values missing. I have nothing that shows info related to the batteries. Is this a Grott service or an integration problem?

I only get 1 device coming through MQTT while the old integration had 2. They may have rolled up into one, since the other integration showed my Shinelink LAN box as ID and now I get the Shinelink RF Stick ID that is in my inverter that connects to the Lanbox

I only see 31 entities on my detected device in Home Assistant Grott and none of them contain the battery related values that are now part opf the MQTT message (I installed the latest beta version after talking to the creator of the Grott).

Can I do something to get these into my Home Assistant system? 

 - Grott values retrieved:
	 -  pvserial             :  ADJ0B000000
	 -  pvstatus             :  1
	 -  pvpowerin            :  964.0
	 -  pv1voltage           :  0.0
	 -  pv1current           :  0.0
	 -  pv1watt              :  0.0
	 -  pv2voltage           :  246.4
	 -  pv2current           :  3.9
	 -  pv2watt              :  964.0
	 -  pv3voltage           :  0.0
	 -  pv3current           :  0.0
	 -  pv3watt              :  0.0
	 -  pv4voltage           :  0.0
	 -  pv4current           :  0.0
	 -  pv4watt              :  0.0
	 -  pvpowerout           :  612.4
	 -  pvfrequentie         :  49.9
	 -  pvgridvoltage        :  241.0
	 -  pvgridcurrent        :  2.6
	 -  pvgridpower          :  612.7
	 -  pvgridvoltage2       :  0.0
	 -  pvgridcurrent2       :  0.0
	 -  pvgridpower2         :  0.0
	 -  pvgridvoltage3       :  0.0
	 -  pvgridcurrent3       :  0.0
	 -  pvgridpower3         :  0.0
	 -  vacrs                :  241.0
	 -  vacst                :  0.0
	 -  vactr                :  0.0
	 -  ptousertotal         :  0.0
	 -  ptogridtotal         :  0.0
	 -  ptoloadtotal         :  758.0
	 -  totworktime          :  9795.5
	 -  pvenergytoday        :  9.5
	 -  pvenergytotal        :  6648.7
	 -  epvtotal             :  7020.2
	 -  epv1today            :  0.0
	 -  epv1total            :  3489.0
	 -  epv2today            :  11.1
	 -  epv2total            :  3531.2
	 -  epv3today            :  0.0
	 -  epv3total            :  0.0
	 -  etousertoday         :  1.2
	 -  etousertotal         :  3517.6
	 -  etogridtoday         :  0.1
	 -  etogridtotal         :  2341.1
	 -  eloadtoday           :  12.8
	 -  eloadtotal           :  8391.6
	 -  deratingmode         :  0
	 -  iso                  :  65530
	 -  dcir                 :  0.0
	 -  dcis                 :  0.0
	 -  dcit                 :  0.0
	 -  gfci                 :  0
	 -  pvtemperature        :  34.6
	 -  pvipmtemperature     :  34.6
	 -  temp3                :  0.0
	 -  temp4                :  0.0
	 -  temp5                :  41.1
	 -  pbusvoltage          :  399.9
	 -  nbusvoltage          :  0.0
	 -  ipf                  :  20000
	 -  realoppercent        :  17
	 -  opfullwatt           :  0.0
	 -  standbyflag          :  0
	 -  faultcode            :  0
	 -  warningcode          :  0
	 -  systemfaultword0     :  0
	 -  systemfaultword1     :  0
	 -  systemfaultword2     :  758
	 -  systemfaultword3     :  758
	 -  systemfaultword4     :  117
	 -  systemfaultword5     :  0
	 -  systemfaultword6     :  0
	 -  systemfaultword7     :  0
	 -  invstartdelaytime    :  60
	 -  bdconoffstate        :  1
	 -  drycontactstate      :  0
	 -  edischrtoday         :  4.5
	 -  edischrtotal         :  1652.3
	 -  echrtoday            :  3.9
	 -  echrtotal            :  1457.4
	 -  eacchrtoday          :  0.0
	 -  eacchrtotal          :  179.4
	 -  priority             :  0
	 -  epsfac               :  0.0
	 -  epsvac1              :  0.0
	 -  epsiac1              :  0.0
	 -  epspac1              :  0.0
	 -  epsvac2              :  0.0
	 -  epsiac2              :  0.0
	 -  epspac2              :  0.0
	 -  epsvac3              :  0.0
	 -  epsiac3              :  0.0
	 -  epspac3              :  0.0
	 -  epspac               :  0.0
	 -  loadpercent          :  0.0
	 -  pf                   :  1000.0
	 -  dcv                  :  0
	 -  bdc1_sysstatemode    :  257
	 -  bdc1_faultcode       :  0
	 -  bdc1_warncode        :  0
	 -  bdc1_vbat            :  212.8
	 -  bdc1_ibat            :  0.9
	 -  bdc1_soc             :  49
	 -  bdc1_vbus1           :  400.4
	 -  bdc1_vbus2           :  203.7
	 -  bdc1_ibb             :  0.6
	 -  bdc1_illc            :  0.6
	 -  bdc1_tempa           :  28.4
	 -  bdc1_tempb           :  0.0
	 -  bdc1_pdischr         :  0.0
	 -  bdc1_pchr            :  201.0
	 -  bdc1_edischrtotal    :  1652.3
	 -  bdc1_echrtotal       :  1457.4
	 -  bdc1_flag            :  3
	 -  bdc2_sysstatemode    :  41
	 -  bdc2_faultcode       :  16
	 -  bdc2_warncode        :  0
	 -  bdc2_vbat            :  2.6
	 -  bdc2_ibat            :  23.7
	 -  bdc2_soc             :  2
	 -  bdc2_vbus1           :  1.6
	 -  bdc2_vbus2           :  0.0
	 -  bdc2_ibb             :  0.0
	 -  bdc2_illc            :  0.1
	 -  bdc2_tempa           :  0.0
	 -  bdc2_tempb           :  5.0
	 -  bdc2_pdischr         :  327680.0
	 -  bdc2_pchr            :  0.0
	 -  bdc2_edischrtotal    :  0.0
	 -  bdc2_echrtotal       :  0.0
	 -  bdc2_flag            :  0
	 -  bms_status           :  1
	 -  bms_error            :  0
	 -  bms_warninfo         :  0
	 -  bms_soc              :  49
	 -  bms_batteryvolt      :  213.0
	 -  bms_batterycurr      :  1.2
	 -  bms_batterytemp      :  39.9
	 -  bms_maxcurr          :  25.0
	 -  bms_deltavolt        :  25.0
	 -  bms_cyclecnt         :  0
	 -  bms_soh              :  100
	 -  bms_constantvolt     :  227.2
	 -  bms_bms_info         :  18880
	 -  bms_packinfo         :  0
	 -  bms_usingcap         :  0
	 -  bms_fw               :  10400
	 -  bms_mcuversion       :  0
	 -  bms_commtype         :  0
 - MQTT jsonmsg: 
		 {"device": "ADJ0B00000", "time": "2023-05-29T16:48:20", "buffered": "no",
		 "values": {"pvserial": "ADJ0B1302U", "pvstatus": 1, "pvpowerin": 9640,
		 "pv1voltage": 0, "pv1current": 0, "pv1watt": 0, "pv2voltage": 2464,
		 "pv2current": 39, "pv2watt": 9640, "pv3voltage": 0, "pv3current": 0,
		 "pv3watt": 0, "pv4voltage": 0, "pv4current": 0, "pv4watt": 0, "pvpowerout":
		 6124, "pvfrequentie": 4994, "pvgridvoltage": 2410, "pvgridcurrent": 26,
		 "pvgridpower": 6127, "pvgridvoltage2": 0, "pvgridcurrent2": 0,
		 "pvgridpower2": 0, "pvgridvoltage3": 0, "pvgridcurrent3": 0, "pvgridpower3":
		 0, "vacrs": 2410, "vacst": 0, "vactr": 0, "ptousertotal": 0, "ptogridtotal":
		 0, "ptoloadtotal": 7580, "totworktime": 70527314, "pvenergytoday": 95,
		 "pvenergytotal": 66487, "epvtotal ": 70202, "epv1today ": 0, "epv1total":
		 34890, "epv2today": 111, "epv2total": 35312, "epv3today": 0, "epv3total": 0,
		 "etousertoday": 12, "etousertotal": 35176, "etogridtoday": 1,
		 "etogridtotal": 23411, "eloadtoday": 128, "eloadtotal": 83916,
		 "deratingmode": 0, "iso": 65530, "dcir": 0, "dcis": 0, "dcit": 0, "gfci": 0,
		 "pvtemperature": 346, "pvipmtemperature": 346, "temp3": 0, "temp4": 0,
		 "temp5": 411, "pbusvoltage": 3999, "nbusvoltage": 0, "ipf": 20000,
		 "realoppercent": 17, "opfullwatt": 0, "standbyflag": 0, "faultcode": 0,
		 "warningcode": 0, "systemfaultword0": 0, "systemfaultword1": 0,
		 "systemfaultword2": 758, "systemfaultword3": 758, "systemfaultword4": 117,
		 "systemfaultword5": 0, "systemfaultword6": 0, "systemfaultword7": 0,
		 "invstartdelaytime": 60, "bdconoffstate": 1, "drycontactstate": 0,
		 "edischrtoday": 45, "edischrtotal": 16523, "echrtoday": 39, "echrtotal":
		 14574, "eacchrtoday": 0, "eacchrtotal": 1794, "priority": 0, "epsfac": 0,
		 "epsvac1": 0, "epsiac1": 0, "epspac1": 0, "epsvac2": 0, "epsiac2": 0,
		 "epspac2": 0, "epsvac3": 0, "epsiac3": 0, "epspac3": 0, "epspac": 0,
		 "loadpercent": 0, "pf": 10000, "dcv": 0, "bdc1_sysstatemode": 257,
		 "bdc1_faultcode": 0, "bdc1_warncode": 0, "bdc1_vbat": 21279, "bdc1_ibat": 9,
		 "bdc1_soc": 49, "bdc1_vbus1": 4004, "bdc1_vbus2": 2037, "bdc1_ibb": 6,
		 "bdc1_illc": 6, "bdc1_tempa": 284, "bdc1_tempb": 0, "bdc1_pdischr": 0,
		 "bdc1_pchr": 2010, "bdc1_edischrtotal": 16523, "bdc1_echrtotal": 14574,
		 "bdc1_flag": 3, "bdc2_sysstatemode": 41, "bdc2_faultcode": 16,
		 "bdc2_warncode": 0, "bdc2_vbat": 259, "bdc2_ibat": 237, "bdc2_soc": 2,
		 "bdc2_vbus1": 16, "bdc2_vbus2": 0, "bdc2_ibb": 0, "bdc2_illc": 1,
		 "bdc2_tempa": 0, "bdc2_tempb": 50, "bdc2_pdischr": 3276800, "bdc2_pchr": 0,
		 "bdc2_edischrtotal": 0, "bdc2_echrtotal": 0, "bdc2_flag": 0, "bms_status":
		 1, "bms_error": 0, "bms_warninfo": 0, "bms_soc": 49, "bms_batteryvolt":
		 21300, "bms_batterycurr": 120, "bms_batterytemp": 399, "bms_maxcurr": 2500,
		 "bms_deltavolt": 2500, "bms_cyclecnt": 0, "bms_soh": 100,
		 "bms_constantvolt": 22720, "bms_bms_info": 18880, "bms_packinfo": 0,
		 "bms_usingcap": 0, "bms_fw": 10400, "bms_mcuversion": 0, "bms_commtype": 0}}
 - Grott MQTT topic used : energy/growatt
 - MQTT message message sent

**Comment by bartpaeleman** [comment]
Ignore, I git it working today.

**Comment by bartpaeleman** [comment]
My temporary workaround is by adding sensors through my MQTT integration.

```
  sensor:
  - name: Home Battery Level
    unique_id: 001_batt_home
    state_topic: energy/growatt
    device_class: battery
    unit_of_measurement: "%"
    value_template: "{{ value_json['values']['bms_soc'] }}"
    icon: mdi:solar-power
```

**Comment by Snux** [comment]
In the grott add on, I changed the type to sph and all the missing battery values came through

**Comment by bartpaeleman** [comment]
You are talking about the grott.ioni file? I was in communication with the creator and he advised me to install the latest beta version with type=min after which I saw some of the values too. The problem is that the HA addon here doesn’t pick up my values since the names are different. My SOC is called bms_soc in the MQTT JSON.

**Comment by muppet3000** [comment]
Hi Both,

Sorry for the delayed response, I've been on holiday. 

Really glad to see that you've got it partially working and some of the values are coming through. If I remember correctly from the old integration @bartpaeleman your system is a `min` system, not an `sph` system, which explains why some of the values are missing.

I appreciate all of the information you've passed above, however, what I would really benefit from is a list of just the values that are missing when you run the integration and what you expect them to map to, that way I can then add them to this integration e.g something like this:

- `bms_soc`:
  - Name: "Statement of Charge"
  - Unit of measurement: "%"
etc.

Once I have those I can add them in for you, if you don't know what all of them should be that's fine, just give me the ones that you know.

**Comment by bartpaeleman** [comment]
Hi @muppet3000, welcome back Chris. 
When I activate the integration these are the values that I see:

<img width="307" alt="image" src="https://github.com/muppet3000/homeassistant-grott/assets/772458/08f85a0d-028b-4c1d-a858-9be117a2e181">
<img width="319" alt="image" src="https://github.com/muppet3000/homeassistant-grott/assets/772458/012f01e6-ff7f-41bb-9e1d-64f882441e82">
<img width="317" alt="image" src="https://github.com/muppet3000/homeassistant-grott/assets/772458/feb5ff6d-cb17-489d-acfb-cf64529cac10">

The values for the battery that you see in teh loist above (which came from activating the BETA docker-version that I was guided towards by the creatoe of the comnpinent, don’t show up. Yes, my sytem must be a MIN indeed, I even had to change the grott.ini file to use this value (upon advise of him) too: invtype = min

**Comment by bartpaeleman** [comment]
@muppet3000, not sure if this is of any help but I used the MQTT values that I get sent through the experimental ledidobe/grott:2.8.2 image with MIN support to extract a few values that I can use.

MQTT Sensors:

`    - name: Home Battery Load
      unique_id: growatt_batteryload
      state_topic: energy/growatt
      device_class: battery
      unit_of_measurement: "%"
      value_template: "{{ value_json['values']['bms_soc'] }}"
      icon: mdi:solar-power

    - name: Battery in vandaag
      unique_id: growatt_battery_in_vandaag
      state_topic: energy/growatt
      device_class: battery
      unit_of_measurement: "kWh"
      value_template: "{{ value_json['values']['echrtoday'] | float / 10 }}"
      icon: mdi:battery-plus-outline

    - name: Battery uit vandaag
      unique_id: growatt_battery_uit_vandaag
      state_topic: energy/growatt
      device_class: battery
      unit_of_measurement: "kWh"
      value_template: "{{ value_json['values']['edischrtoday'] | float / 10 }}"
      icon: mdi:battery-minus-outline

    - name: Battery in totaal
      unique_id: growatt_battery_in_totaal
      state_topic: energy/growatt
      state_class: total_increasing
      device_class: battery
      unit_of_measurement: "kWh"
      value_template: "{{ value_json['values']['echrtotal'] }}"
      icon: mdi:battery-plus-variant

    - name: Battery uit totaal
      unique_id: growatt_battery_uit_totaal
      state_topic: energy/growatt
      state_class: total_increasing
      device_class: battery
      unit_of_measurement: "kWh"
      value_template: "{{ value_json['values']['edischrtotal'] }}"
      icon: mdi:battery-minus-variant

    - name: Growatt - Datalogger serienr
      unique_id: growatt_datlogger_serial
      state_topic: energy/growatt
      value_template: "{{ value_json['device'] }}" 
      icon: mdi:select-inverse

    - name: Growatt - Datum
      unique_id: growatt_date
      state_topic: energy/growatt
      value_template: "{{ as_timestamp(strptime(value_json['time'], '%Y-%m-%dT%H:%M:%S')) | timestamp_custom('%d-%m-%Y') }}" 
      icon: mdi:calendar

    - name: Growatt - Tijd
      unique_id: growatt_time
      state_topic: energy/growatt
      value_template: "{{ as_timestamp(strptime(value_json['time'], '%Y-%m-%dT%H:%M:%S')) | timestamp_custom('%H:%M:%S') }}" 
      icon: mdi:clock-digital

    - name: Growatt - Status
      unique_id: growatt_status
      state_topic: energy/growatt
      value_template: >
        {% if (value_json['values']['pvstatus'] | int == 0) %}
          Wachtend
        {% elif (value_json['values']['pvstatus'] | int == 1) %}
          Normaal
        {% elif (value_json['values']['pvstatus'] | int == 2) %}
          Fout
        {% else %}
          Onbekend
        {% endif %}
      icon: mdi:power-settings

    - name: Growatt - Vermogen uit
      unique_id: growatt_string1_watt
      state_topic: energy/growatt
      value_template: "{{ value_json['values']['pvgridpower'] | float / 10000 }}" 
      device_class: power
      unit_of_measurement: "kW"

    - name: Growatt - Voltage uit
      unique_id: growatt_string1_voltage
      state_topic: energy/growatt
      value_template: "{{ value_json['values']['pvgridvoltage'] | float / 10 }}" 
      device_class: voltage
      unit_of_measurement: "V"

    - name: Growatt - Stroom uit
      unique_id: growatt_string1_current
      state_topic: energy/growatt
      value_template: "{{ value_json['values']['pvgridcurrent'] | float / 10 }}" 
      device_class: current
      unit_of_measurement: "A"

    - name: Growatt - Input wattage (Actueel)
      unique_id: growatt_actual_input_power
      state_topic: energy/growatt
      value_template: "{{ value_json['values']['pvpowerin'] | float / 10000 }}" 
      device_class: power
      unit_of_measurement: "kW"

    - name: Growatt - Output wattage (Actueel)
      unique_id: growatt_actual_output_power
      state_topic: energy/growatt
      value_template: "{{ value_json['values']['pvpowerout'] | float / 10000 }}" 
      device_class: power
      unit_of_measurement: "kW"
 
    - name: Growatt - Gridfrequentie
      unique_id: growatt_grid_frequency
      state_topic: energy/growatt
      value_template: "{{ value_json['values']['pvfrequentie'] | float / 100 }}" 
      unit_of_measurement: "Hz"
      icon: mdi:waveform
 
    - name: Growatt - Fase voltage
      unique_id: growatt_phase_voltage
      state_topic: energy/growatt
      value_template: "{{ value_json['values']['pvgridvoltage'] | float / 10 }}" 
      device_class: voltage
      unit_of_measurement: "V"

    - name: Growatt - Opgewekte energie (Vandaag)
      unique_id: growatt_generated_energy_today
      state_topic: energy/growatt
      value_template: "{{ value_json['values']['pvenergytoday'] | float / 10 }}" 
      device_class: energy
      unit_of_measurement: "kWh"
      icon: mdi:solar-power

    - name: Growatt - Opgewekte energie (Totaal)
      unique_id: growatt_generated_energy_total
      state_topic: energy/growatt
      value_template: "{{ value_json['values']['pvenergytotal'] | float / 10 }}" 
      device_class: energy
      state_class: total_increasing
      unit_of_measurement: "kWh"
      icon: mdi:solar-power

    - name: Growatt - Inverter temperatuur
      unique_id: growatt_inverer_temperature
      state_topic: energy/growatt
      value_template: "{{ value_json['values']['pvtemperature'] | float / 10 }}" 
      device_class: temperature
      unit_of_measurement: "°C" 
`

<img width="510" alt="image" src="https://github.com/muppet3000/homeassistant-growatt_server_api/assets/772458/c5c2995e-4c78-4871-b155-be04bc0d0071">

**Comment by muppet3000** [comment]
@bartpaeleman Please could you try pulling the 'main' version of this integration via HACS and then letting me know if the missing battery values now appear?

**Comment by bartpaeleman** [comment]
@muppet3000 will do and will check in the morning to let you know. Do I keep the type=min in my grott.ini that I had to put there for the beta version?

**Comment by muppet3000** [comment]
Yes, don't change anything with your Grott configuration, simply just pull the 'main' version of this integration through hacs and it should pull in the battery values (you can confirm this by looking in the 'devices' section of HA).
The names of the values should appear from this list: https://github.com/muppet3000/homeassistant-grott/commit/c7f0a06d1dcdec21a45b3e02a4fde7545183e088 (I've tried to keep them consistent with other sensor names from other types)

**Comment by bartpaeleman** [comment]
Done, now restarting and will check tomorrow morning once my MQTT values come in if I see a change.

**Comment by muppet3000** [comment]
Do your MQTT values not get pushed every few minutes from the Shine Wifi stick? The new sensors should appear as soon as there is a push to MQTT.

**Comment by bartpaeleman** [comment]
@muppet3000 I have a ShineLink-X RF stick, not WiFi and it seems to be pretty unstable.The values aren’t really sent within minutes, it can take up to 20 minutes or more before I see something. I had replaced my docker container with the latest version yesterday (thinking that was what you expected) initially and didn’t see much. This morning I have rebuilt the container with the image that I had before 2.8.2) but am still waiting after 19 minutes to see values appear,

**Comment by bartpaeleman** [comment]
They just passed through my GROTT log but nothing visible in the integration yet. My MQTTlistener caught these values in growatt/energy:

`{
  "device": "ADJ0B1302U",
  "time": "2023-06-21T07:40:03",
  "buffered": "no",
  "values": {
    "pvserial": "ADJ0B1302U",
    "pvstatus": 1,
    "pvpowerin": 4048,
    "pv1voltage": 2561,
    "pv1current": 7,
    "pv1watt": 1954,
    "pv2voltage": 2574,
    "pv2current": 8,
    "pv2watt": 2094,
    "pv3voltage": 0,
    "pv3current": 0,
    "pv3watt": 0,
    "pv4voltage": 0,
    "pv4current": 0,
    "pv4watt": 0,
    "pvpowerout": 4620,
    "pvfrequentie": 4998,
    "pvgridvoltage": 2337,
    "pvgridcurrent": 20,
    "pvgridpower": 4623,
    "pvgridvoltage2": 0,
    "pvgridcurrent2": 0,
    "pvgridpower2": 0,
    "pvgridvoltage3": 0,
    "pvgridcurrent3": 0,
    "pvgridpower3": 0,
    "vacrs": 2337,
    "vacst": 0,
    "vactr": 0,
    "ptousertotal": 4092,
    "ptogridtotal": 0,
    "ptoloadtotal": 10020,
    "totworktime": 74390734,
    "pvenergytoday": 19,
    "pvenergytotal": 71314,
    "epvtotal ": 75426,
    "epv1today ": 2,
    "epv1total": 37299,
    "epv2today": 1,
    "epv2total": 38127,
    "epv3today": 0,
    "epv3total": 0,
    "etousertoday": 23,
    "etousertotal": 35853,
    "etogridtoday": 1,
    "etogridtotal": 24295,
    "eloadtoday": 50,
    "eloadtotal": 89378,
    "deratingmode": 0,
    "iso": 65530,
    "dcir": 0,
    "dcis": 0,
    "dcit": 0,
    "gfci": 0,
    "pvtemperature": 348,
    "pvipmtemperature": 348,
    "temp3": 0,
    "temp4": 0,
    "temp5": 432,
    "pbusvoltage": 3740,
    "nbusvoltage": 0,
    "ipf": 20000,
    "realoppercent": 12,
    "opfullwatt": 0,
    "standbyflag": 0,
    "faultcode": 0,
    "warningcode": 0,
    "systemfaultword0": 0,
    "systemfaultword1": 0,
    "systemfaultword2": 592,
    "systemfaultword3": 592,
    "systemfaultword4": 27,
    "systemfaultword5": 0,
    "systemfaultword6": 0,
    "systemfaultword7": 0,
    "invstartdelaytime": 60,
    "bdconoffstate": 1,
    "drycontactstate": 0,
    "edischrtoday": 25,
    "edischrtotal": 18642,
    "echrtoday": 0,
    "echrtotal": 16248,
    "eacchrtoday": 0,
    "eacchrtotal": 1835,
    "priority": 0,
    "epsfac": 0,
    "epsvac1": 0,
    "epsiac1": 0,
    "epspac1": 0,
    "epsvac2": 0,
    "epsiac2": 0,
    "epspac2": 0,
    "epsvac3": 0,
    "epsiac3": 0,
    "epspac3": 0,
    "epspac": 0,
    "loadpercent": 0,
    "pf": 10000,
    "dcv": 0,
    "bdc1_sysstatemode": 513,
    "bdc1_faultcode": 0,
    "bdc1_warncode": 701,
    "bdc1_vbat": 20445,
    "bdc1_ibat": 9,
    "bdc1_soc": 14,
    "bdc1_vbus1": 3731,
    "bdc1_vbus2": 1894,
    "bdc1_ibb": 2,
    "bdc1_illc": 2,
    "bdc1_tempa": 315,
    "bdc1_tempb": 0,
    "bdc1_pdischr": 1880,
    "bdc1_pchr": 0,
    "bdc1_edischrtotal": 18642,
    "bdc1_echrtotal": 16248,
    "bdc1_flag": 1,
    "bdc2_sysstatemode": 42,
    "bdc2_faultcode": 11,
    "bdc2_warncode": 0,
    "bdc2_vbat": 306,
    "bdc2_ibat": 286,
    "bdc2_soc": 2,
    "bdc2_vbus1": 14,
    "bdc2_vbus2": 0,
    "bdc2_ibb": 0,
    "bdc2_illc": 1,
    "bdc2_tempa": 0,
    "bdc2_tempb": 50,
    "bdc2_pdischr": 3276800,
    "bdc2_pchr": 0,
    "bdc2_edischrtotal": 0,
    "bdc2_echrtotal": 0,
    "bdc2_flag": 0,
    "bms_status": 4,
    "bms_error": 0,
    "bms_warninfo": 0,
    "bms_soc": 14,
    "bms_batteryvolt": 20480,
    "bms_batterycurr": 65496,
    "bms_batterytemp": 434,
    "bms_maxcurr": 2500,
    "bms_deltavolt": 2500,
    "bms_cyclecnt": 0,
    "bms_soh": 100,
    "bms_constantvolt": 22720,
    "bms_bms_info": 18880,
    "bms_packinfo": 0,
    "bms_usingcap": 0,
    "bms_fw": 10275,
    "bms_mcuversion": 0,
    "bms_commtype": 0
  }
} `

**Comment by muppet3000** [comment]
Just to be clear, you're getting some values in the Grott integration in HA, just not the new battery ones I added?

**Comment by bartpaeleman** [comment]
I am going to remove my current one (says v1.01) and will disable my custom MQTT values, then pull the main again and reboot. 
![image](https://github.com/muppet3000/homeassistant-grott/assets/772458/364b5f53-221b-4c2d-8449-868e8f16ceea)

Update will follow.

**Comment by muppet3000** [comment]
Ok, that's the issue then, you need to re-download and then choose "main" from the dropdown:
1. ![image](https://github.com/muppet3000/homeassistant-grott/assets/10612068/9bce12a3-cac7-4b23-bb09-6dcb2ecbe7d9)

2. 
![image](https://github.com/muppet3000/homeassistant-grott/assets/10612068/d0459d5e-0e07-4b59-bd27-18b500683f4e)

**Comment by bartpaeleman** [comment]
Something weird going on then. I did as you wrote, grabbed the MAIN but after reboot:
![image](https://github.com/muppet3000/homeassistant-grott/assets/772458/592dcfc3-9974-4e8b-9b70-a963fad4ce4c)

Going to remove both integrations ()including the other Growatt one), reboot and then do it all over again.

**Comment by bartpaeleman** [comment]
Got it Chris, now I will wait and see what appears:
![image](https://github.com/muppet3000/homeassistant-grott/assets/772458/9a906446-03fd-46f6-92dc-71f13cdf41d4)

**Comment by bartpaeleman** [comment]
Looking good Chris!!
![image](https://github.com/muppet3000/homeassistant-grott/assets/772458/d6b96f06-60b7-4fde-83b4-1353d520c184)
![image](https://github.com/muppet3000/homeassistant-grott/assets/772458/04269e8e-cb84-45f9-a567-ac090497243a)
![image](https://github.com/muppet3000/homeassistant-grott/assets/772458/b7e37885-7635-4d27-a327-3993ce24df6a)

**Comment by muppet3000** [comment]
Woop woop! I'm just working through some stuff over on #15 but hopefully we're mostly fixed then!

**Comment by bartpaeleman** [comment]
Enjoy the Paypal beers Chris ;)

**Comment by muppet3000** [comment]
Thanks @bartpaeleman 🍺 

I'll close this ticket off when I do the next release.

**Comment by muppet3000** [comment]
Released in v1.1.0
